src/inmanta/plugins.py

```diff
@@ -380,6 +379,9 @@ def resolve_type(self, plugin: "Plugin", resolver: Namespace) -> inmanta_type.Ty
             return self._resolved_type
 
         if not isinstance(self.type_expression, str):
+            if typing_inspect.is_union_type(self.type_expression) and not typing.get_args(self.type_expression):
+                # If typing.Union is not subscripted, isinstance(self.type_expression, type) evaluates to False.
+                raise TypingException(None, f"Union type must be subscripted, got {self.type_expression}")
```

Comment:
I'm hesitant about the exception type. The one raised further down is a `RuntimeException`. I think they should at least be consistent.

Reply:
I think it's fine to use this one for both, but @wouterdb probably knows best.